src/cli/cli_uart.cpp

```diff
@@ -237,25 +237,52 @@ otError Uart::ProcessCommand(void)
 int Uart::Output(const char *aBuf, uint16_t aBufLength)
 {
     OT_CLI_UART_OUTPUT_LOCK();
-    uint16_t remaining = kTxBufferSize - mTxLength;
-    uint16_t tail;
+    uint16_t sent = 0;
 
-    if (aBufLength > remaining)
+    while (aBufLength > 0)
     {
-        aBufLength = remaining;
-    }
+        uint16_t remaining = kTxBufferSize - mTxLength;
+        uint16_t tail;
+        uint16_t sendLength = aBufLength;
 
-    for (int i = 0; i < aBufLength; i++)
-    {
-        tail            = (mTxHead + mTxLength) % kTxBufferSize;
-        mTxBuffer[tail] = *aBuf++;
-        mTxLength++;
+        if (sendLength > remaining)
+        {
+            sendLength = remaining;
+        }
+
+        for (int i = 0; i < sendLength; i++)
```

Comment:
->?
```cpp
for (uint16_t i = 0; i < sendLength; i++)
```

Reply:
I'll update it … but for reference, this was the original code:
https://github.com/openthread/openthread/blob/master/src/cli/cli_uart.cpp#L248